sdks/python/apache_beam/io/gcp/pubsub.py

```diff
@@ -146,15 +180,26 @@ def __init__(self, topic):
     # we are using this coder explicitly for portability reasons of PubsubIO
     # across implementations in languages.
     self.coder = coders.BytesCoder()
-    self.topic = topic
+    self.full_topic = topic
+
+    # Validate topic name.
```

Comment:
The `match` handling and `project` + `self.topic_name` assignment below are copy-pasted from `_PubSubPayloadSource`. I suggest to extract two new methods, e.g., `parse_topic` and `parse_subscription`, and use these here and in `_PubSubPayloadSource`.